### PR TITLE
[magentatv] Adapt to new Telekom OAuth flow (required to retrieve the userId)

### PR DESCRIPTION
**Pull request by markus7017**
This PR adapted to the new OAuth flow, current version is not able to retrieve the userId from the authenticated account.

### CODE REVIEW

**Comment by markus7017** [comment]
@openhab/add-ons-maintainers  Could someone pick-up this small PR? Telekom changed the auth flow, so new users are no longer able to signin with the current version included with the distro. This PR fixes the problem and should be easy to review

File: bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java [hunk 1]
```diff
@@ -185,7 +189,7 @@ private void initializeThing() {
     }
 
     /**
-     * This routine is called every time the Thing configuration has been changed.
+     * This routine is called every time the Thing configuration has been changed (e.g. PaperUI)
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java:192]: There is no such thing as a PaperUI anymore.
```suggestion
     * This routine is called every time the Thing configuration has been changed.
```

File: bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java [hunk 2]
```diff
@@ -78,124 +93,210 @@ public String getUserId(String accountName, String accountPassword) throws Magen
...
+        return result;
     }
 
     private Properties initHttpHeader() {
         Properties httpHeader = new Properties();
-        httpHeader.setProperty(HEADER_USER_AGENT, OAUTH_USER_AGENT);
-        httpHeader.setProperty(HEADER_ACCEPT, "*/*");
-        httpHeader.setProperty(HEADER_LANGUAGE, "de-de");
-        httpHeader.setProperty(HEADER_CACHE_CONTROL, "no-cache");
+        // httpHeader.setProperty(HttpHeader.USER_AGENT.toString(), OAUTH_USER_AGENT);
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java:267]: Can this line be deleted?

File: bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/binding/binding.xml [hunk 3]
```diff
@@ -5,5 +5,6 @@
 
 	<name>MagentaTV Binding</name>
 	<description>This is the binding for MagentaTV receivers</description>
+	<author>Markus Michels</author>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/binding/binding.xml:8]: The author tag is deprecated, we don't use it anymore.
```suggestion
```

File: bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -21,7 +21,6 @@
 		</channel-groups>
 
 
-		<representation-property>macAddress</representation-property>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/thing/thing-types.xml:24]: What is the new representation property?

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thanks @markus7017, just some small comments.

File: bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java:192]: reference removed

File: bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java (hunk 2 above)
Comment (markus7017) [reply bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java:267]: revmoed

File: bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/binding/binding.xml (hunk 3 above)
Comment (markus7017) [reply bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/binding/binding.xml:8]: removed

File: bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 4 above)
Comment (markus7017) [reply bundles/org.openhab.binding.magentatv/src/main/resources/OH-INF/thing/thing-types.xml:24]: hmm, good catch, maybe accidentally deleted

**Comment by markus7017** [comment]
@kaikreuzer review changes applied

**Review by kaikreuzer — APPROVED** [review]
Perfect, thank you!

**Pull request merged**